src/main/java/edu/ucsb/cs156/organic/controllers/CoursesController.java

```diff
@@ -132,4 +136,39 @@ public Iterable<Staff> getStaff(
         return courseStaff;
     }
 
+    @Operation(summary = "Update a course")
+    @PreAuthorize("hasAnyRole('ROLE_ADMIN', 'ROLE_INSTRUCTOR')")
+    @PutMapping("/update")
+    public Course updateCourse(
+            @Parameter(name = "courseId") @RequestParam Long courseId,
+            @RequestBody @Valid Course incoming) throws JsonProcessingException {
+        
+        User u = getCurrentUser().getUser();
+        Course course = courseRepository.findById(courseId)
+                .orElseThrow(() -> new EntityNotFoundException(Course.class, courseId.toString()));
+        Iterable<Staff> courseStaff = courseStaffRepository.findByCourseId(courseId);
+
+        //check if the user is a staff member of the course
+        boolean isStaff = false;
+        for (Staff staff : courseStaff) {
+            //log.info("staffId: {}, uId: {}, {}", staff.getGithubId(), u.getGithubId(), staff.getGithubId() == u.getGithubId());
+            if (staff.getGithubId().equals(u.getGithubId())) {
+                isStaff = true;
+                break;
+            }
+        }
+
+        if(!isStaff)
+            throw new StaffNotFoundException(u.getGithubId(), courseId);
```

Comment:
I think it may be better to use an instance of an already existing exception, namely: 

```
import org.springframework.security.access.AccessDeniedException;
...

Then:
```
throw new AccessDeniedException(String.format(
   "user with github id %s is not on the staff of course with id %d",
    u.getGithubId(), course.getId()));
```

Then you don't need to define a new exception type.

That should mapped not to 404 not found,  but to 403, access denied.

Reply:
See https://github.com/ucsb-cs156-f23/proj-organic-f23-6pm-1/pull/40